### PR TITLE
fix: package clobber and `__pycache__` removal issues

### PR DESCRIPTION
**Pull request by wolfv**

### CODE REVIEW

File: examples/flask-hello-world/pixi.toml [hunk 1]
```diff
@@ -14,4 +14,3 @@ start = "python -m flask run --port=5050"
 
 [dependencies]
 python = "3.11.*"
-flask = "2.*"
```
Comment (ruben-arts) [inline examples/flask-hello-world/pixi.toml:17]: I guess this was accidental?

File: src/install.rs [hunk 2]
```diff
@@ -320,35 +321,15 @@ async fn remove_package_from_environment(
...
-            Err(e) if e.kind() == ErrorKind::NotFound => {
-                // Simply ignore if the file is already gone.
-            }
-            Err(e) => {
-                return Err(e).into_diagnostic().wrap_err(format!(
-                    "failed to delete {}",
-                    paths.relative_path.display()
-                ))
-            }
+    match unlink_package(target_prefix, package).await {
```
Comment (ruben-arts) [inline src/install.rs:324]: So this now takes care of all the todo's?

File: src/install.rs [hunk 3]
```diff
@@ -320,35 +321,15 @@ async fn remove_package_from_environment(
...
+        Ok(_) => Ok(()),
+        Err(err) => {
+            println!("\n\n\nFailed to remove package: {}", err);
+            miette::bail!(
+                "Failed to remove package {}: {}",
+                package.repodata_record.file_name,
+                err
+            );
```
Comment (ruben-arts) [inline src/install.rs:332]: I guess bailing might even be a lot. Wouldn't it be better to return an error which could then be reported as a warning perhaps. Or is this a really big issue?

If so what should the user do when this happens?

File: src/install.rs (hunk 2 above)
Comment (wolfv) [reply src/install.rs:324]: Nope, one of the todo's just moved to rattler, the other one is taken care of :)

File: src/install.rs [hunk 4]
```diff
@@ -320,35 +321,15 @@ async fn remove_package_from_environment(
...
-                    "failed to delete {}",
-                    paths.relative_path.display()
-                ))
-            }
+    match unlink_package(target_prefix, package).await {
+        Ok(_) => Ok(()),
+        Err(err) => {
+            println!("\n\n\nFailed to remove package: {}", err);
+            miette::bail!(
+                "Failed to remove package {}: {}",
```
Comment (baszalmstra) [inline src/install.rs:329]: With errors, please start a sentence with a lowercase letter to make concatenated errors look better.

**Comment by baszalmstra** [comment]
@wolfv Can you bring this PR up to date? `main` is currently tracking rattler `v0.15.0`

**Comment by wolfv** [comment]
yep, will fix it up!

**Comment by wolfv** [comment]
I fixed the merge issues, but this still needs the PR to go in in Rattler ...

File: examples/flask-hello-world/pixi.toml (hunk 1 above)
Comment (wolfv) [reply examples/flask-hello-world/pixi.toml]: undone!

File: src/install.rs (hunk 3 above)
Comment (wolfv) [reply src/install.rs]: Apparently, `bail!` returns early with an error (which is what we want in this case).

https://docs.rs/miette/latest/miette/macro.bail.html

File: src/install.rs (hunk 4 above)
Comment (wolfv) [reply src/install.rs]: 👍

File: src/install.rs [hunk 5]
```diff
@@ -321,35 +318,7 @@ async fn remove_package_from_environment(
...
-
-    // Remove the conda-meta file
-    let conda_meta_path = target_prefix.join("conda-meta").join(format!(
-        "{}-{}-{}.json",
-        package.repodata_record.package_record.name.as_normalized(),
-        package.repodata_record.package_record.version,
-        package.repodata_record.package_record.build
-    ));
-    tokio::fs::remove_file(conda_meta_path)
+    unlink_package(target_prefix, package)
```
Comment (ruben-arts) [inline src/install.rs:321]: Lol nice job on the LOC removal

File: src/install.rs [hunk 6]
```diff
@@ -29,7 +29,7 @@ pub async fn execute_transaction(
     let package_cache = PackageCache::new(cache_dir.join("pkgs"));
 
     // Create an install driver which helps limit the number of concurrent filesystem operations
-    let install_driver = InstallDriver::default();
+    let install_driver = InstallDriver::new(100, Some(prefix_records));
```
Comment (ruben-arts) [inline src/install.rs:32]: Was this  100 a guess or is it based on something?

**Review by ruben-arts — APPROVED** [review]

File: src/install.rs (hunk 6 above)
Comment (wolfv) [reply src/install.rs:32]: It's the value we are currently using as default (so it's the same as before).

**Comment by wolfv** [comment]
Btw. I would like to move most of the code here to `rattler` at some point ... into the `Transaction`, so that we have a `Transacation::execute(...)` function.

**Pull request merged**